src/hydra-queue-runner/build-remote.cc

```diff
@@ -175,24 +175,301 @@ StorePaths reverseTopoSortPaths(const std::map<StorePath, ValidPathInfo> & paths
     return sorted;
 }
 
+std::pair<Path, AutoCloseFD> openLogFile(const std::string & logDir, const StorePath & drvPath)
+{
+    string base(drvPath.to_string());
+    auto logFile = logDir + "/" + string(base, 0, 2) + "/" + string(base, 2);
+
+    createDirs(dirOf(logFile));
+
+    AutoCloseFD logFD = open(logFile.c_str(), O_CREAT | O_TRUNC | O_WRONLY, 0666);
+    if (!logFD) throw SysError("creating log file ‘%s’", logFile);
+
+    return {std::move(logFile), std::move(logFD)};
+}
+
+void handshake(Machine::Connection & conn, unsigned int repeats)
```

Comment:
I think it would either be better to rename the function (`nixServeV1Handshake` or something like that) or move it into a namespace/class to make its use more obvious

Reply:
Yup' fair point. I've moved all that new logic to its own namespace to make things clearer